libraries/AP_Camera/AP_Camera.h

```diff
@@ -88,6 +92,8 @@ class AP_Camera {
     AP_Int16        _servo_off_pwm;     // PWM value to move servo to when shutter is deactivated
     uint8_t         _trigger_counter;   // count of number of cycles shutter has been held open
     AP_Relay       *_apm_relay;         // pointer to relay object from the base class Relay.
+    AP_Int8         _auto_only;         // if 1: trigger by distance only if in AUTO mode.
+    AP_Int8         _control_mode;      // flight mode number
```

Comment:
Why is mode an AP_Int8?

Reply:
I could have used uint8_t   , for both, it's just that many parameters seems to use AP_Int8 ? - should I change it ?

Reply:
AP_Int8 is for parameters. The mode isn't a parameter so there isn't any reason to use an AP_Int8 object for it.